feat(validation dependencies)

LailaVanRooijen opened the pull request:
Added dependencies for validation

https://www.baeldung.com/spring-boot-bean-validation

astraliskye commented on backend/pom.xml:
From the article, it appears we only need the `spring-boot-starter-validation` dependency. Where do `hibernate-validator` and `jakarta.validation-api` come from?

astraliskye commented on backend/src/main/java/com/mystery/project/exception/GlobalExceptionHandler.java:
This seems to be working correctly, but I notice when no request body is provided, an `HttpMessageNotReadableException` will occur and the client gets a bad message response: `{
	"timestamp": "2025-02-08T23:22:49.287+00:00",
	"status": 400,
	"error": "Bad Request",
	"path": "/api/v1"
}`

astraliskye replied:
Unsure if the bad request response is intended over the validation style response

LailaVanRooijen commented:
I don't remember why i added the other two dependencies, but they don't seem to be neccesary so i removed them.

I added a method in the globalExceptionHandler that catches the requests with an empty body

And added problemdetail to the handle validation method
